spring-cloud-zuul-ratelimit-core/src/main/java/com/marcosbarbero/cloud/autoconfigure/zuul/ratelimit/config/repository/RedisRateLimiter.java

```diff
@@ -62,26 +62,19 @@ protected void calcRemainingQuota(final Long quota, final Long refreshInterval,
     private Long calcRemaining(Long limit, Long refreshInterval, long usage,
                                String key, Rate rate) {
         rate.setReset(SECONDS.toMillis(refreshInterval));
-        Long current = 0L;
+
         try {
-            current = redisTemplate.opsForValue().increment(key, usage);
-            // Redis returns the value of key after the increment, check for the first increment, and the expiration time is set
-            if (current != null && current.equals(usage)) {
-                handleExpiration(key, refreshInterval);
+            Long current = 0L;
+            Boolean present = this.redisTemplate.opsForValue().setIfAbsent(key, usage, refreshInterval, SECONDS);
+            if (Boolean.FALSE.equals(present)) {
+                // Key already exists, increment
+              current = this.redisTemplate.opsForValue().increment(key, usage);
             }
+            return Math.max(-1, limit - (current != null ? current : 0L));
```

Comment:
I was thinking, maybe we instead of adding this line in here we could just replace the `return limit;` with
```
return Math.max(-1, limit - (current != null ? current : 0L));
```

WDYT?

Reply:
I agree that it is better to have a single return place, but unfortunately I deleted my local branch for this PR and I don't know if I can update the code. If you create a separate branch for this commit I will change the code.

Reply:
You can check out this PR locally. This guide may be helpful for you https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/checking-out-pull-requests-locally

Reply:
Thank you for the guide, but I can't push to this PR or create a new branch on your repo. I can created another PR just for this change.  Is this ok?

Reply:
That's also fine :)